src/HeadProvider.js

```diff
@@ -59,10 +59,15 @@ export default class HeadProvider extends React.Component {
       }
       headTags.push(tagNode);
     },
+
+    whitelist: this.props.whitelist || [],
   };
 
   componentDidMount() {
-    const ssrTags = document.head.querySelectorAll(`[data-rh=""]`);
+    const whitelist = this.state.whitelist.length
+      ? `,${this.state.whitelist.join(`,`)}`
+      : ``;
+    const ssrTags = document.head.querySelectorAll(`[data-rh=""]${whitelist}`);
```

Comment:
Maybe I'm misunderstanding, but isn't the point of this PR to remove the need for `data-rh`? That is, if a `whitelist` is provided, shouldn't the querySelectorAll look for only the whitelist and not `data-rh`?

e.g.
```js
const query = this.state.whitelist.length ? this.state.whitelist.join(',') : '[data-rh=""]';
const ssrTags = document.head.querySelectorAll(query);
```

And now having written this out, heh this makes me think maybe we should just accept a generic "react-head-managed-selector" string (single) instead of an array, call it `selector`. So something like:

```js
const query = this.state.selector || '[data-rh=""]';
const ssrTags = document.head.querySelectorAll(selector);
```

With usage:

```jsx
<HeadProvider headTags={headTags} selector='title,[name="description"],[property^="og:"]'}>
  <App />
</HeadProvider>
```

Sorry for the churn (too much context switching for me today! 😅 )

Reply:
Never mind! I see you want to only disable `data-rh` when the selector is _not_ in the whitelist. Makes sense! 👍

Reply:
Haha it's hard sometimes I know^^
Exactly, I thought it's nice to be able to only opt out of data-rh for certain elements and leave the rest alone and automatically managed by the plugin...
Any thoughts? You prefer an additional option or something to fully disable `data-rh`?

Reply:
> I thought it's nice to be able to only opt out of data-rh for certain elements and leave the rest alone and automatically managed by the plugin...

Right, I think it makes sense and this is an elegant solution. I wonder if it's obvious enough in the README though? Like, on first glance, it's not clear to me that if a tag isn't covered by the whitelist that it will fallback to providing a `data-rh` attribute -- maybe we should add a note about that in the docs?